store/store_helper.go

```diff
@@ -0,0 +1,57 @@
+package store
+
+import (
+	types "github.com/cosmos/cosmos-sdk/store/types"
+)
+
+// GetAndDecode gets and decodes key and returns it. Returns nil if key doesn't exist.
+func GetAndDecode[T any](store types.KVStore, dec func([]byte) (T, error), key []byte) (T, error) {
+	var res T
+	bz := store.Get(key)
+	if bz == nil {
+		return res, nil
+	}
+	resp, err := dec(bz)
+	if err != nil {
+		return resp, err
+	}
+	return resp, err
+}
+
+// GetAndDecodeWithBool gets and decodes key and returns it. Returns false if key doesn't exist.
+func GetAndDecodeWithBool[T any](store types.KVStore, dec func([]byte) (T, bool), key []byte) (T, bool) {
+	var res T
+	bz := store.Get(key)
+	if len(bz) == 0 {
+		return res, false
+	}
+	resp, boolval := dec(bz)
+	return resp, boolval
+}
+
+// StoreAPI is a wrapper around the store's KVStore to provide more safe key management and better ease-of-use.
+type StoreAPI struct {
+	types.KVStore
+}
+
+// NewStoreAPI returns a new KVStore.
+func NewStoreAPI(store types.KVStore) StoreAPI {
+	return StoreAPI{
+		KVStore: store,
+	}
+}
+
+// Set stores the value under the given key.
+func (store StoreAPI) Set(key []byte, value []byte) {
+	store.KVStore.Set(key, value)
+}
+
+// Get returns the raw bytes stored under the given key. Returns nil when key does not exist.
+func (store StoreAPI) Get(key []byte) []byte {
+	return store.KVStore.Get(key)
+}
+
+// Delete deletes the value stored under the given key, if it exists.
+func (store StoreAPI) Delete(key []byte) {
+	store.KVStore.Delete(key)
```

Comment:
Why do we need to define all of these methods? Since we embed a `KVStore` directly and we're not doing anything different, then redefining each method is pointless.

Reply:
I totally agree with you, but its just that I addressed this review comment here: https://github.com/cosmos/cosmos-sdk/pull/13446#discussion_r987948880
I'm waiting for people to point out this, it's this way just to have all the methods under the same hood of store helper.

Reply:
If you people think it's of no use, I'm up for doing the required changes.

Reply:
I'm not following exactly. As of right now, `StoreAPI` isn't doing anything different or special. It's not doing what @julienrbrt referenced [here](https://github.com/axelarnetwork/axelar-core/blob/b7ada310c210e7e5c7af3995fd14eef4c67dee14/utils/store.go#L40-L43).

So as of right now, `StoreAPI` is pointless.